### ISSUE TITLE
Ubuntu 16.04 agents lack most of the .NET Core SDKs/runtimes that MacOS has

### ISSUE THREAD

**Issue by AArnott** [Microsoft/azure-pipelines-image-generation#234]
The Ubuntu 16.04 queue has just one .NET Core SDK and runtime installed:

```
2018-09-13T13:28:52.4444459Z dotnet --info
2018-09-13T13:28:52.4624946Z [command]/bin/bash --noprofile --norc /home/vsts/work/_temp/1ea70124-025c-4876-9202-bef679e724f9.sh
2018-09-13T13:28:52.5412377Z .NET Core SDK (reflecting any global.json):
2018-09-13T13:28:52.5428311Z  Version:   2.1.300
2018-09-13T13:28:52.5441069Z  Commit:    adab45bf0c
2018-09-13T13:28:52.5447305Z 
2018-09-13T13:28:52.5460245Z Runtime Environment:
2018-09-13T13:28:52.5472976Z  OS Name:     ubuntu
2018-09-13T13:28:52.5487147Z  OS Version:  16.04
2018-09-13T13:28:52.5500285Z  OS Platform: Linux
2018-09-13T13:28:52.5690674Z  RID:         ubuntu.16.04-x64
2018-09-13T13:28:52.5706959Z  Base Path:   /opt/hostedtoolcache/dncs/2.1.300/x64/sdk/2.1.300/
2018-09-13T13:28:52.5724092Z 
2018-09-13T13:28:52.5743542Z Host (useful for support):
2018-09-13T13:28:52.5757454Z   Version: 2.1.0
2018-09-13T13:28:52.5771089Z   Commit:  caa7b7e2ba
2018-09-13T13:28:52.5777633Z 
2018-09-13T13:28:52.5791317Z .NET Core SDKs installed:
2018-09-13T13:28:52.5805322Z   2.1.300 [/opt/hostedtoolcache/dncs/2.1.300/x64/sdk]
2018-09-13T13:28:52.5812284Z 
2018-09-13T13:28:52.5826490Z .NET Core runtimes installed:
2018-09-13T13:28:52.5840804Z   Microsoft.AspNetCore.All 2.1.0 [/opt/hostedtoolcache/dncs/2.1.300/x64/shared/Microsoft.AspNetCore.All]
2018-09-13T13:28:52.5854665Z   Microsoft.AspNetCore.App 2.1.0 [/opt/hostedtoolcache/dncs/2.1.300/x64/shared/Microsoft.AspNetCore.App]
2018-09-13T13:28:52.5868765Z   Microsoft.NETCore.App 2.1.0 [/opt/hostedtoolcache/dncs/2.1.300/x64/shared/Microsoft.NETCore.App]
2018-09-13T13:28:52.5875422Z 
2018-09-13T13:28:52.5890995Z To install additional .NET Core runtimes or SDKs:
2018-09-13T13:28:52.5904195Z   https://aka.ms/dotnet-download
```

In contrast, MacOS has a great complement of preinstalled .NET Core runtimes and SDKs (below), which building projects that target various versions, and testing those projects when running on several distinct runtimes:

```
2018-09-13T13:28:07.3421460Z dotnet --info
2018-09-13T13:28:07.3582410Z [command]/bin/bash --noprofile --norc /Users/vsts/agent/2.140.0/work/_temp/bc53c063-8a64-4a41-a4cf-dc0cf2a95039.sh
2018-09-13T13:28:08.1586450Z .NET Core SDK (reflecting any global.json):
2018-09-13T13:28:08.1601210Z  Version:   2.1.401
2018-09-13T13:28:08.1616170Z  Commit:    91b1c13032
2018-09-13T13:28:08.1623500Z 
2018-09-13T13:28:08.1638270Z Runtime Environment:
2018-09-13T13:28:08.1653280Z  OS Name:     Mac OS X
2018-09-13T13:28:08.1667890Z  OS Version:  10.13
2018-09-13T13:28:08.1682410Z  OS Platform: Darwin
2018-09-13T13:28:08.2378540Z  RID:         osx.10.13-x64
2018-09-13T13:28:08.2395100Z  Base Path:   /Users/vsts/.dotnet/sdk/2.1.401/
2018-09-13T13:28:08.2440360Z 
2018-09-13T13:28:08.2455030Z Host (useful for support):
2018-09-13T13:28:08.2469480Z   Version: 2.1.3
2018-09-13T13:28:08.2483370Z   Commit:  124038c13e
2018-09-13T13:28:08.2491130Z 
2018-09-13T13:28:08.2505130Z .NET Core SDKs installed:
2018-09-13T13:28:08.2519410Z   1.0.1 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2533760Z   1.0.4 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2549250Z   1.1.4 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2564060Z   1.1.5 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2578610Z   1.1.7 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2593210Z   1.1.8 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2607230Z   1.1.9 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2621710Z   1.1.10 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2635820Z   2.0.0 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2650450Z   2.0.3 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2664580Z   2.1.2 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2678600Z   2.1.4 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2692710Z   2.1.100 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2706880Z   2.1.101 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2721340Z   2.1.102 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2735930Z   2.1.103 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2750360Z   2.1.104 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2764810Z   2.1.105 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2778980Z   2.1.200 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2793180Z   2.1.201 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2807260Z   2.1.202 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2821610Z   2.1.300 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2835740Z   2.1.301 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2849860Z   2.1.302 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2864100Z   2.1.400 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2878120Z   2.1.401 [/Users/vsts/.dotnet/sdk]
2018-09-13T13:28:08.2885390Z 
2018-09-13T13:28:08.2899520Z .NET Core runtimes installed:
2018-09-13T13:28:08.2913830Z   Microsoft.AspNetCore.All 2.1.0 [/Users/vsts/.dotnet/shared/Microsoft.AspNetCore.All]
2018-09-13T13:28:08.2928430Z   Microsoft.AspNetCore.All 2.1.1 [/Users/vsts/.dotnet/shared/Microsoft.AspNetCore.All]
2018-09-13T13:28:08.2942870Z   Microsoft.AspNetCore.All 2.1.2 [/Users/vsts/.dotnet/shared/Microsoft.AspNetCore.All]
2018-09-13T13:28:08.2957630Z   Microsoft.AspNetCore.All 2.1.3 [/Users/vsts/.dotnet/shared/Microsoft.AspNetCore.All]
2018-09-13T13:28:08.2972120Z   Microsoft.AspNetCore.App 2.1.0 [/Users/vsts/.dotnet/shared/Microsoft.AspNetCore.App]
2018-09-13T13:28:08.2987020Z   Microsoft.AspNetCore.App 2.1.1 [/Users/vsts/.dotnet/shared/Microsoft.AspNetCore.App]
2018-09-13T13:28:08.3001510Z   Microsoft.AspNetCore.App 2.1.2 [/Users/vsts/.dotnet/shared/Microsoft.AspNetCore.App]
2018-09-13T13:28:08.3016350Z   Microsoft.AspNetCore.App 2.1.3 [/Users/vsts/.dotnet/shared/Microsoft.AspNetCore.App]
2018-09-13T13:28:08.3030810Z   Microsoft.NETCore.App 1.0.4 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3045180Z   Microsoft.NETCore.App 1.0.5 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3059810Z   Microsoft.NETCore.App 1.0.7 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3074730Z   Microsoft.NETCore.App 1.0.8 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3089190Z   Microsoft.NETCore.App 1.0.9 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3103670Z   Microsoft.NETCore.App 1.0.10 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3118230Z   Microsoft.NETCore.App 1.0.11 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3133400Z   Microsoft.NETCore.App 1.0.12 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3148100Z   Microsoft.NETCore.App 1.1.1 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3162790Z   Microsoft.NETCore.App 1.1.2 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3177770Z   Microsoft.NETCore.App 1.1.4 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3192530Z   Microsoft.NETCore.App 1.1.5 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3206970Z   Microsoft.NETCore.App 1.1.6 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3221430Z   Microsoft.NETCore.App 1.1.7 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3236220Z   Microsoft.NETCore.App 1.1.8 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3250770Z   Microsoft.NETCore.App 1.1.9 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3265310Z   Microsoft.NETCore.App 2.0.0 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3279790Z   Microsoft.NETCore.App 2.0.3 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3294080Z   Microsoft.NETCore.App 2.0.5 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3307850Z   Microsoft.NETCore.App 2.0.6 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3322650Z   Microsoft.NETCore.App 2.0.7 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3337140Z   Microsoft.NETCore.App 2.0.9 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3351970Z   Microsoft.NETCore.App 2.1.0 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3367620Z   Microsoft.NETCore.App 2.1.1 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3382340Z   Microsoft.NETCore.App 2.1.2 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3397610Z   Microsoft.NETCore.App 2.1.3 [/Users/vsts/.dotnet/shared/Microsoft.NETCore.App]
2018-09-13T13:28:08.3405190Z 
2018-09-13T13:28:08.3419610Z To install additional .NET Core runtimes or SDKs:
2018-09-13T13:28:08.3435970Z   https://aka.ms/dotnet-download
2018-09-13T13:28:08.3487060Z ##[section]Finishing: Show dotnet SDK info
```

Can the Linux queues receive the same set as MacOS does, please?

**Comment by dakale** [comment]
Hey @AArnott thanks for reporting. I added all the dotnet sdks + runtimes that Microsoft packages for Ubuntu 16. Here's a list of whats now on all Hosted Ubuntu 16 machines: 

```
2018-09-21T15:31:00.2495407Z dotnet --info
2018-09-21T15:31:06.8615559Z .NET Core SDK (reflecting any global.json):
2018-09-21T15:31:06.8633622Z  Version:   2.1.402
2018-09-21T15:31:06.8651569Z  Commit:    3599f217f4
2018-09-21T15:31:06.8660452Z 
2018-09-21T15:31:06.8678576Z Runtime Environment:
2018-09-21T15:31:06.8697029Z  OS Name:     ubuntu
2018-09-21T15:31:06.8714838Z  OS Version:  16.04
2018-09-21T15:31:06.8732463Z  OS Platform: Linux
2018-09-21T15:31:07.1060590Z  RID:         ubuntu.16.04-x64
2018-09-21T15:31:07.1685985Z  Base Path:   /usr/share/dotnet/sdk/2.1.402/
2018-09-21T15:31:07.1704488Z 
2018-09-21T15:31:07.1723349Z Host (useful for support):
2018-09-21T15:31:07.1741316Z   Version: 2.1.4
2018-09-21T15:31:07.1759619Z   Commit:  85255dde3e
2018-09-21T15:31:07.1768506Z 
2018-09-21T15:31:07.1786777Z .NET Core SDKs installed:
2018-09-21T15:31:07.1804419Z   1.0.4 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.1823408Z   1.1.4 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.1843265Z   1.1.5 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.1867559Z   1.1.7 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.1885399Z   1.1.8 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.1903072Z   1.1.9 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.1921342Z   1.1.10 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.1938841Z   2.0.0 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.1956884Z   2.0.3 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.1979337Z   2.1.2 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.1997208Z   2.1.4 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.2014531Z   2.1.100 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.2032787Z   2.1.101 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.2050295Z   2.1.102 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.2068609Z   2.1.103 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.2086362Z   2.1.104 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.2104093Z   2.1.105 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.2121730Z   2.1.200 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.2139290Z   2.1.201 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.2157134Z   2.1.202 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.2174915Z   2.1.402 [/usr/share/dotnet/sdk]
2018-09-21T15:31:07.2183916Z 
2018-09-21T15:31:07.2201490Z .NET Core runtimes installed:
2018-09-21T15:31:07.2219135Z   Microsoft.AspNetCore.All 2.1.4 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
2018-09-21T15:31:07.2237361Z   Microsoft.AspNetCore.App 2.1.4 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
2018-09-21T15:31:07.2255944Z   Microsoft.NETCore.App 1.0.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2274097Z   Microsoft.NETCore.App 1.0.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2292171Z   Microsoft.NETCore.App 1.0.8 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2310175Z   Microsoft.NETCore.App 1.0.9 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2330451Z   Microsoft.NETCore.App 1.0.10 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2348565Z   Microsoft.NETCore.App 1.0.11 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2366685Z   Microsoft.NETCore.App 1.0.12 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2384924Z   Microsoft.NETCore.App 1.1.2 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2403262Z   Microsoft.NETCore.App 1.1.4 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2423000Z   Microsoft.NETCore.App 1.1.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2442913Z   Microsoft.NETCore.App 1.1.6 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2462802Z   Microsoft.NETCore.App 1.1.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2481482Z   Microsoft.NETCore.App 1.1.8 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2499369Z   Microsoft.NETCore.App 1.1.9 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2517595Z   Microsoft.NETCore.App 2.0.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2535711Z   Microsoft.NETCore.App 2.0.3 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2559215Z   Microsoft.NETCore.App 2.0.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2577346Z   Microsoft.NETCore.App 2.0.6 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2595745Z   Microsoft.NETCore.App 2.0.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2613779Z   Microsoft.NETCore.App 2.0.9 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
2018-09-21T15:31:07.2631958Z   Microsoft.NETCore.App 2.1.4 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

**Comment by AArnott** [comment]
@dakale I just noticed that `2.1.300` isn't listed above. That's the original .NET Core 2.1.0 release, and should be on this list. It is compatible with Ubuntu 16, as I have it on my docker image.

Can you please add that, and any other missing versions, since it seemed you intended to cover them all?

Thanks.

**Comment by dakale** [comment]
2.1.300 was for some reason not in the package repo, but I cant figure out if that means theres some known incompatibility (seems like no). I installed that one, and any others that arent in the package repos manually, and heres the new output (not live yet though, still testing):

```
 Runtime Environment:
  OS Name:     ubuntu
  OS Version:  16.04
  OS Platform: Linux
  RID:         ubuntu.16.04-x64
  Base Path:   /usr/share/dotnet/sdk/2.1.402/

 Host (useful for support):
   Version: 2.1.4
   Commit:  85255dde3e

 .NET Core SDKs installed:
   1.0.4 [/usr/share/dotnet/sdk]
   1.1.4 [/usr/share/dotnet/sdk]
   1.1.5 [/usr/share/dotnet/sdk]
   1.1.7 [/usr/share/dotnet/sdk]
   1.1.8 [/usr/share/dotnet/sdk]
   1.1.9 [/usr/share/dotnet/sdk]
   1.1.10 [/usr/share/dotnet/sdk]
   2.0.0 [/usr/share/dotnet/sdk]
   2.0.3 [/usr/share/dotnet/sdk]
   2.1.2 [/usr/share/dotnet/sdk]
   2.1.4 [/usr/share/dotnet/sdk]
   2.1.100 [/usr/share/dotnet/sdk]
   2.1.101 [/usr/share/dotnet/sdk]
   2.1.102 [/usr/share/dotnet/sdk]
   2.1.103 [/usr/share/dotnet/sdk]
   2.1.104 [/usr/share/dotnet/sdk]
   2.1.105 [/usr/share/dotnet/sdk]
   2.1.200 [/usr/share/dotnet/sdk]
   2.1.201 [/usr/share/dotnet/sdk]
   2.1.202 [/usr/share/dotnet/sdk]
   2.1.300 [/usr/share/dotnet/sdk]
   2.1.301 [/usr/share/dotnet/sdk]
   2.1.302 [/usr/share/dotnet/sdk]
   2.1.400 [/usr/share/dotnet/sdk]
   2.1.401 [/usr/share/dotnet/sdk]
   2.1.402 [/usr/share/dotnet/sdk]

 .NET Core runtimes installed:
   Microsoft.AspNetCore.All 2.1.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
   Microsoft.AspNetCore.All 2.1.1 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
   Microsoft.AspNetCore.All 2.1.2 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
   Microsoft.AspNetCore.All 2.1.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
   Microsoft.AspNetCore.All 2.1.4 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
   Microsoft.AspNetCore.App 2.1.0 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
   Microsoft.AspNetCore.App 2.1.1 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
   Microsoft.AspNetCore.App 2.1.2 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
   Microsoft.AspNetCore.App 2.1.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
   Microsoft.AspNetCore.App 2.1.4 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
   Microsoft.NETCore.App 1.0.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.0.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.0.8 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.0.9 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.0.10 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.0.11 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.0.12 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.1.2 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.1.4 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.1.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.1.6 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.1.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.1.8 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 1.1.9 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 2.0.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 2.0.3 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 2.0.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 2.0.6 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 2.0.7 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 2.0.9 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 2.1.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 2.1.1 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 2.1.2 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 2.1.3 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
   Microsoft.NETCore.App 2.1.4 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

Can you help me understand the usecase that requires 2.1.300 specifically? Or is it just cross testing against multiple versions?

 For my validation I do: 
```bash
echo "{\"sdk\": { \"version\": \"$sdk\" }}" > global.json
    dotnet new console
    dotnet restore
    dotnet build
```
for each sdk. Does this match the functionality you would expect or is there another scenario I should test against. 

I would be glad to extend my testing to cover any specific builds you may have

**Comment by AArnott** [comment]
Thanks, @dakale. 

The use case is described here: https://github.com/dotnet/sdk/issues/2546 
The 2.1.300 version specifically comes from the fact that that's what all my repos have pinned because it was the official release of .NET Core 2.1 which came out with a bunch of perf fixes. In fact it's the most recent public minor release available -- 2.2 is still in preview and everything since 2.1 has been just servicing fixes. This means there's no compelling need for me to go update all my repos from 2.1.300 to use a newer SDK. But I want them to build in CI on Azure Pipelines as easily as they built on AppVeyor, which included 2.1.300 along with every other public release.

Cross testing across multiple versions is very interesting for *runtimes*, but not for SDKs. I only need my code to build with a single SDK, but then I'll test it, typically on the latest .NET Core runtime for 1.0, 1.1, 2.0, and 2.1. 

Your validation step looks good. It seems to cover the scenario that I care about. But FWIW so long as the SDK shows up under `dotnet --info`, it has never failed for me, so I'd personally be satisfied to just see it in the list and assume it works.
